test/validation/api/classification/odp_classification_basic.c

```diff
@@ -52,6 +52,43 @@ static void classification_test_create_cos(void)
 	odp_queue_destroy(queue);
 }
 
+static void classification_test_create_cos_max_common(odp_bool_t stats)
+{
+	odp_cls_cos_param_t cls_param;
+	odp_cls_capability_t capa;
+
+	CU_ASSERT_FATAL(odp_cls_capability(&capa) == 0);
+
+	int num = capa.max_cos;
+
+	if (stats && (int)capa.max_cos_stats < num)
+		num = capa.max_cos_stats;
+
+	odp_cos_t cos[num];
+
+	for (int i = 0; i < num; i++) {
+		odp_cls_cos_param_init(&cls_param);
+		cls_param.action = ODP_COS_ACTION_DROP;
+		cls_param.stats_enable = stats;
+
+		cos[i] = odp_cls_cos_create(NULL, &cls_param);
+		CU_ASSERT(cos[i] != ODP_COS_INVALID);
```

Comment:
For easier debugging, I'd suggest to break the loop after the first create fails and print values of i and num with ODPH_ERR(). Destroy loop would then run up to num created.